dcrd.go

```diff
@@ -82,12 +82,28 @@ func dcrdMain() error {
 	//
 	// A limit of 1.8 GiB is used as a baseline, with an increase to the limit
 	// when the UTXO cache max size has been increased over the default.
+	//
+	// In addition, the system administrator may override limit to increase it
+	// further by setting the GOMEMLIMIT environment variable.  In order to
+	// prevent extremely poor performance due to a limit that is unreasonably
+	// low, the environment variable will be ignored when it is configured with
+	// a value less than the minimum required amount as determined per the
+	// previous description.
 	const memLimitBase = (18 * (1 << 30)) / 10 // 1.8 GiB
 	softMemLimit := int64(memLimitBase)
 	if cfg.UtxoCacheMaxSize > defaultUtxoCacheMaxSize {
 		extra := int64(cfg.UtxoCacheMaxSize) - defaultUtxoCacheMaxSize
 		softMemLimit += extra * (1 << 20)
 	}
+	goMemLimit := debug.SetMemoryLimit(-1)
+	if goMemLimit < softMemLimit {
+		dcrdLog.Infof("Ignoring GOMEMLIMIT=%q since it is too low and would "+
+			"lead to extremely poor performance",
+			humanizeBytes(goMemLimit))
+	}
+	if os.Getenv("GOMEMLIMIT") != "" && goMemLimit > softMemLimit {
```

Comment:
you've effectively already checked if the variable is set. If it is, it will be something other than the max int64, so no need to check for the os.Getenv here.

Reply:
Yep.  That's why.

Reply:
Updated to use `os.LookupEnv` instead.